File: SPA/Models/DTO/Order.cs
using MongoDB.Bson;
using MongoDB.Bson.Serialization.Attributes;
using Newtonsoft.Json;
using System;
using System.Collections.Generic;
using System.Linq;
using System.Threading.Tasks;

namespace SPA.Models.DTO
{
    public enum OrderStatus
    {
        New,
        InProgress,
        OnHold,
        Done
    }
    public class Order : OrderDto
    {
        [BsonId]
        public ObjectId _id { get; set; }       
      
     
        public new string OId { get => _id.ToString(); set => _ = value; }

    }

    public class OrderDto: CarHistory
    {
        [BsonIgnore]
        [JsonProperty("_id")]
        public string OId { get; set; }
        public string UserOId { get; set; }
        public string CarOId { get; set; }

    }
    public class CarHistory
    {
        public DateTime StartDate { get; set; }
        public DateTime? EndDate { get; set; }
        public decimal Cost { get; set; }
        public List<Part> Parts { get; set; }
        public string Comment { get; set; }
        public OrderStatus OrderStatus { get; set; }
    }
    public class Part
    {
        public string PartName { get; set; }
        public string Brand { get; set; }
        public string CatalogNumber { get; set; }
        public int Qty { get; set; }        
    }
}

File: SPA/ClientApp/src/app/orders/Order.ts
import { CarHistory, OrderStatus, Part } from "../Customers/Car";

export class Order extends CarHistory {
    oId: string;
    carOId: string;
    userOId:string;
 

    constructor(oId: string, carOId: string, startDate:Date, cost:number,comment:string,orderStatus:OrderStatus,userOId:string,endDate?:Date, parts?:Part[] ) {
        super(startDate,cost,comment,orderStatus,endDate,parts);
        this.oId = oId;
        this.carOId = carOId;
        this.userOId = userOId;
    }

}




